armbalancer.go

```diff
@@ -72,16 +72,43 @@ func New(opts Options) http.RoundTripper {
 	}
 
 	if opts.TransportFactory == nil {
-		opts.TransportFactory = newRecyclableTransport
+		opts.TransportFactory = make(map[string]func(id int, parent *http.Transport, host string, port string, recycleThreshold, minReqsBeforeRecycle int64) http.RoundTripper)
+		opts.TransportFactory[strings.ToLower(host+":"+port)] = newRecyclableTransport
```

Comment:
at this point the options struct isn't very intuitive since one can pass a set of transport factories __and__ a host/port.

We haven't released 1.0 yet, so we can make a breaking change to clean this up. What do you think about using the functional options pattern to allow each allowed hostname to be defined with or without a customized http transport?